internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -160,6 +160,17 @@ func resourceKubernetesCluster() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"node_os_channel_upgrade": {
+				Type:     pluginsdk.TypeString,
+				Optional: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(managedclusters.NodeOSUpgradeChannelNodeImage),
+					string(managedclusters.NodeOSUpgradeChannelSecurityPatch),
+					string(managedclusters.NodeOSUpgradeChannelUnmanaged),
+					string(managedclusters.NodeOSUpgradeChannelNone),
```

Comment:
We don't typically expose values like `None` in the provider - some docs about that [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/schema-design-considerations.md#the-none-value-or-similar). This should default to `None` if the property has been removed after having previously been set.
```suggestion
```